src/main/java/uk/gov/hmcts/ccd/security/filters/ServiceAuthFilter.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.ccd.security.filters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.filter.OncePerRequestFilter;
+import uk.gov.hmcts.ccd.security.exception.UnauthorizedException;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.exceptions.ServiceException;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ServiceAuthFilter extends OncePerRequestFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ServiceAuthFilter.class);
+
+    public static final String AUTHORISATION = "ServiceAuthorization";
+
+    private final List<String> authorisedServices;
+    private final AuthTokenValidator authTokenValidator;
+
+    public ServiceAuthFilter(AuthTokenValidator authTokenValidator, List<String> authorisedServices) {
+        this.authTokenValidator = authTokenValidator;
+        if (authorisedServices.isEmpty()) {
+            throw new IllegalArgumentException("Must have at least one service defined");
+        }
```

Comment:
do we need to throw an exception here? if the list of authorised services is empty the request simply won't be allowed

Reply:
Filter bean creation itself fails if authorised list is not configured (empty)